### ISSUE TITLE
remove_button plugin bug: append expects html_container to be a string

### ISSUE THREAD

**Issue by dk-teknologisk-osv** [brianreavis/selectize.js#2065]
I did:

* [x] Search for if my issue has already been submitted
* [x] Make sure I'm reporting something precise that needs to be fixed
* [x] Give my issue a descriptive and concise title
* [ ] Create a *minimal* working example on JsFiddle or Codepen
	(or gave a link to a demo on the Selectize docs)
* [x] Indicate *precise* steps to reproduce in *numbers* and the result,
	  like below

Steps to reproduce:

1. Make a render "item" function that returns a DOM element and not a string

Expected result:

It shall work as the documentation states that

> Custom rendering functions. Each function should accept two arguments: data and escape, and must return HTML (string or DOM element) with a single root element. 

Actual result:

I get a crash:

> TypeError: html_container.search is not a function

here

https://github.com/selectize/selectize.js/blob/e3f2e0b4aa251375bc21b5fcd8ca7d374a921f08/src/plugins/remove_button/plugin.js#L49

Basically the function needs to handle html_container to be either string or a DOM element.

EDIT:

Something like this should do:

```
var append = function(html_container, html_element) {
        if (typeof html_container === 'string') {
          var pos = html_container.search(/(<\/[^>]+>\s*)$/);
          return html_container.substring(0, pos) + html_element + html_container.substring(pos);
        } else if (html_container instanceof Element) {
          html_container.append($(html_element)[0]);
          return html_container;
        }
};
```
